src/Core/BasePlugin.php

```diff
@@ -175,6 +182,19 @@ public function getClassPath(): string
         return $path . 'src' . DIRECTORY_SEPARATOR;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function getTemplatePath(): string
+    {
+        if ($this->templatePath) {
+            return $this->templatePath;
+        }
+        $path = $this->getPath();
+
+        return $path . 'templates' . DIRECTORY_SEPARATOR;
```

Comment:
Should this method set `$this->templatePath` so we don't repeat work the 2nd+ time this method is called?

Reply:
Done